contracts/fio.address/fio.address.cpp

```diff
@@ -809,6 +809,130 @@ namespace fioio {
+            
+            uint32_t domain_expiration = get_now_plus_one_year();
+            domains.emplace(actor, [&](struct domain &d) {
+                d.id = domains.available_primary_key();;
+                d.name = fa.fiodomain;
+                d.domainhash = domainHash;
+                d.expiration = domain_expiration;
+                d.account = actor.value;
+            });
+
+            auto key_iter = accountmap.find(actor.value);
+
+            fio_400_assert(key_iter != accountmap.end(), "owner", to_string(actor.value),
+                           "Owner is not bound in the account map.", ErrorActorNotInFioAccountMap);
+
+            vector <tokenpubaddr> pubaddresses;
+            tokenpubaddr t1;
+            t1.public_address = key_iter->clientkey;
+            t1.token_code = "FIO";
+            t1.chain_code = "FIO";
+            pubaddresses.push_back(t1);
+
+            fionames.emplace(actor, [&](struct fioname &a) {
+                a.id = fionames.available_primary_key();;
+                a.name = fa.fioaddress;
+                a.addresses = pubaddresses;
+                a.namehash = string_to_uint128_hash(fa.fioaddress.c_str());;
+                a.domain = fa.fiodomain;
+                a.domainhash = domainHash;
+                a.expiration = 4294967295; //Sunday, February 7, 2106 6:28:15 AM GMT+0000 (Max 32 bit expiration)
+                a.owner_account = actor.value;
+                a.bundleeligiblecountdown = getBundledAmount();
+            });
+            
+            const uint128_t endpoint_hash = string_to_uint128_hash(REGISTER_FIO_DOMAIN_ADDRESS_ENDPOINT);
+
+            auto fees_by_endpoint = fiofees.get_index<"byendpoint"_n>();
+            auto fee_iter = fees_by_endpoint.find(endpoint_hash);
+            fio_400_assert(fee_iter != fees_by_endpoint.end(), "endpoint_name", REGISTER_FIO_DOMAIN_ADDRESS_ENDPOINT,
+                           "FIO fee not found for endpoint", ErrorNoEndpoint);
+
+            const uint64_t reg_amount = fee_iter->suf_amount;
+            const uint64_t fee_type = fee_iter->type;
+
+            fio_400_assert(fee_type == 0, "fee_type", to_string(fee_type),
+                           "unexpected fee type for endpoint register_fio_address, expected 0",
+                           ErrorNoEndpoint);
+
+            fio_400_assert(max_fee >= (int64_t) reg_amount, "max_fee", to_string(max_fee),
+                           "Fee exceeds supplied maximum.",
+                           ErrorMaxFeeExceeded);
+
+            fio_fees(actor, asset(reg_amount, FIOSYMBOL), REGISTER_FIO_DOMAIN_ADDRESS_ENDPOINT);
+            processbucketrewards(tpid, reg_amount, get_self(), actor);
+
+            if (REGDOMADDRAM > 0) {
+                action(
+                        permission_level{SYSTEMACCOUNT, "active"_n},
+                        "eosio"_n,
+                        "incram"_n,
+                        std::make_tuple(actor, REGDOMADDRAM)
```

Comment:
Please confirm with Pawel if it is okay to reuse REGDOMADDRAM or if this call should have its own fee.

Reply:
This is a new constant

Reply:
Ah, right. looks like it is set to 5120. Looks good. Closing.